src/environmentd/tests/sql.rs

```diff
@@ -1822,3 +1822,22 @@ fn test_load_generator() -> Result<(), Box<dyn Error>> {
 
     Ok(())
 }
+
+#[test]
+fn test_cancel_on_closed_connection() -> Result<(), anyhow::Error> {
```

Comment:
This doesn't fail on main, because it relies on a race condition.

Reply:
That's ok, we just need to demonstrate it fails once, and can't get it to fail after the change. To do that, Can you put the below `server.runtime.block_on(async { ... ` thing in a loop and wait for it to crash? Might be quick because it's just making a new connection, not a new server, each iteration. If that doesn't do it within 10s, try making a bunch of threads that all reconnect in a loop. I suspect that'll be hard or annoying because of rust async stuff.